### PR TITLE
Refactor TOC

### PR DESCRIPTION
**Pull request by natke**
Move welcome into the Overview section. (Changes to the welcome page will be coming soon)
Move conceptual docs into a Concepts section
Re-title some of the TOC headings

Preview link: https://review.docs.microsoft.com/en-us/quantum/?view=qsharp-preview&branch=pr-en-us-550

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/TOC.yml](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/b40ac9ea6397fb58ee2369c505f7625a3e9872b3/articles/TOC.yml) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/eb0fe707-1307-402f-bbaa-84627cd1637a_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/TOC.yml](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/1bbdecf11862c509adf60979c8faa2b0693af3bb/articles/TOC.yml) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/57324cda-c348-43b2-9fca-462c2cb60deb_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/TOC.yml](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/d054a16786c64dc30d2946a8dff323d7f7eef8ba/articles/TOC.yml) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/4f52a3f0-53eb-4855-9a80-add36b862197_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/TOC.yml](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/7d2236f91d29e9f115b79b51e808f6a0610ba395/articles/TOC.yml) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/2a5e2211-68d5-4b25-8bc8-a7176ddc5206_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by gillenhaalb** [comment]
My two cents: I feel like the "Quantum Computing Concepts" section should stay completely separate from the other "Concepts" which are primarily about Q# and how to apply it . That could also help avoid what Chris mentioned (e.g. people inquiring about how to do a tensor product in Q#).

File: articles/TOC.yml [hunk 1]
```diff
@@ -12,25 +10,31 @@
...
-- name: Glossary
-  href: Glossary.md
-- name: For more information
-  href: for-more-info.md
-- name: Release Notes
-  topicHref: relnotes/index.md
-  href: relnotes/toc.yml
-- name: License
-  href: license.md
+- name: Concepts
```
Comment (cgranade) [inline articles/TOC.yml]: I'd suggest moving the techniques chapter above the QC concepts chapter, if only to help address the issue feedback we've gotten that the concepts guide currently feels like an intimidating prereq.

File: articles/TOC.yml [hunk 2]
```diff
@@ -12,25 +10,31 @@
...
+    href: concepts/toc.yml
+  - name: Quantum development techniques
+    topicHref: techniques/index.md
+    href: techniques/toc.yml
+  - name: Q# programming language
+    topicHref: language/index.md
+    href: language/toc.yml
+  - name: How to host a quantum machine
+    href: machines/toc.yml
+  - name: Q# libraries
```
Comment (cgranade) [inline articles/TOC.yml]: Similarly, I'd suggest moving this above `machines/*`, as I would expect (and hope!) that using the libraries is a more basic concept than customizing classical host programs.

File: api/toc.yml [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
-  - name: Trace simulator 
-    href: https://docs.microsoft.com/dotnet/api/microsoft.quantum.simulation.simulators.qctracesimulators
-  - name: Quantum chemistry
-    href: https://docs.microsoft.com/dotnet/api/microsoft.quantum.chemistry
-- name: Python libraries
-  href: https://docs.microsoft.com/python/qsharp    
\ No newline at end of file
+  - name: Q# libraries
+    href: qsharp/toc.yml
+  - name: C# libraries
```
Comment (cgranade) [inline api/toc.yml]: Do we want to take the opportunity to emphasize that these are .NET libraries, not just C#?

Comment (natke) [reply api/toc.yml]: Yep, can do.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards

**A minimum Acrolinx score of 80 is required.**

Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:

| Article | Acrolinx<br>Score | Lowest<br>Category<br>Score | Issues | Scorecard | Processed |
|---------|:--------------:|:--------------------:|:------:|:---------:|:---------:|
| [articles/TOC.yml](https://github.com/MicrosoftDocs/quantum-docs-pr/blob/1420e8163f8b3c5f49356e62c26b27fe45dfb394/articles/TOC.yml) | 100 | 100 | 0 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/843de52f-29d0-439a-80a4-feea0aa5f73b_report.html) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**

- [Install Acrolinx locally for VSCode](https://review.docs.microsoft.com/en-us/help/contribute/contribute-acrolinx-vscode?branch=master)
- [False positives or issues](https://aka.ms/acrolinxbug)

**Comment by natke** [comment]
@gillenhaalb I can definitely see this. We do want to preserve the content model where we only have one Concepts section. We can keep iterating though!

**Pull request merged**